### PR TITLE
feat(cli): add manifest delete command

### LINKED ISSUE #154: feat(cli): manifest delete command
Add a `delete` sub-command to the existing `manifest` command that allows users to remove flags from their manifest file via the CLI. This complements the `add` sub-command and provides a complete flag lifecycle management workflow.

Users should be able to specify a flag name and have it safely removed from the manifest file with proper validation and error handling.

## Acceptance Criteria

- [ ] Implement `openfeature manifest delete` command
- [ ] Command requires flag name as mandatory parameter
- [ ] Command validates that the flag exists before attempting deletion
- [ ] Use temp file pattern for safe I/O operations:
  - [ ] Write updated manifest to temporary file
  - [ ] Complete all I/O operations on temp file  
  - [ ] Only overwrite original manifest if no errors occur
- [ ] Command provides clear success/error feedback
- [ ] Command follows existing CLI patterns and conventions
- [ ] Add appropriate help text and usage examples
- [ ] Include unit tests for the new functionality
- [ ] Handle edge cases (empty manifest, non-existent flags, etc.)

## Notes

- Should integrate with existing manifest command structure
- Follow same temp file pattern used in manifest add command
- Consider adding confirmation prompt for destructive operations
- This builds on the manifest management feature set alongside the add command

### PR DESCRIPTION
**Pull request by reckziegelwilliam**
Implement manifest delete subcommand to allow users to remove flags from their manifest file via the CLI. This complements the existing add subcommand and provides complete flag lifecycle management.

Features:
- Command validates flag exists before deletion
- Uses atomic temp file pattern for safe I/O operations
- Provides clear success/error feedback
- Follows existing CLI patterns and conventions
- Includes comprehensive unit tests (9 test cases)
- Handles edge cases (empty manifest, non-existent flags, etc.)

Fixes #154

## This PR

- Adds new `openfeature manifest delete <flag-name>` subcommand
- Validates that the specified flag exists in the manifest before deletion
- Uses the existing atomic temp file pattern (`manifest.Write()`) for safe I/O operations
- Provides clear success and error messages using pterm
- Follows all existing CLI patterns and conventions (PreRunE, config helpers, filesystem usage, etc.)
- Includes debug logging consistent with `manifest add` command
- Adds comprehensive test coverage with 9 test functions (464 lines) covering all edge cases

**Files Added:**
- `internal/cmd/manifest_delete.go` - implements the delete command (91 lines)
- `internal/cmd/manifest_delete_test.go` - comprehensive test suite (464 lines)

**Files Modified:**
- `internal/cmd/manifest.go` - registers the delete subcommand (1 line added)
- `internal/config/flags.go` - adds `AddManifestDeleteFlags()` helper for consistency (4 lines added)

### Related Issues

Fixes #154

### Notes

- The implementation reuses the existing `manifest.Write()` function which already implements the atomic temp file pattern (write to temp → close → rename), ensuring no data corruption occurs
- All validation logic follows the same patterns as `manifest add` and `manifest list` commands
- Error handling covers all edge cases: missing manifest file, empty manifest, non-existent flag
- Success messages include both flag name and manifest path for clear user feedback
- Command integrates seamlessly with the existing `--manifest` flag for custom manifest paths
- Debug logging added to match the pattern used in `manifest add` for consistency

### Follow-up Tasks

None - all acceptance criteria from issue #154 have been fully implemented and tested.

### How to test

**1. Automated Tests**
# Run all tests
go test ./...

# Run specific manifest delete tests
go test ./internal/cmd/... -v -run TestManifestDelete

# Verify build
go build ./cmd/openfeature**2. Manual Testing**

Create a test manifest:
cat > test-flags.json << 'EOF'
{
  "$schema": "https://raw.githubusercontent.com/open-feature/cli/refs/heads/main/schema/v0/flag-manifest.json",
  "flags": {
    "feature-a": {
      "flagType": "boolean",
      "defaultValue": true,
      "description": "First feature flag"
    },
    "feature-b": {
      "flagType": "string",
      "defaultValue": "test",
      "description": "Second feature flag"
    },
    "feature-c": {
      "flagType": "integer",
      "defaultValue": 42,
      "description": "Third feature flag"
    }
  }
}
EOFTest basic deletion:
# Delete a flag
./openfeature manifest delete feature-a --manifest test-flags.json

# Verify the flag was removed
cat test-flags.json
# Should show only feature-b and feature-c remain

# Delete another flag
./openfeature manifest delete feature-b --manifest test-flags.json

# Delete the last flag
./openfeature manifest delete feature-c --manifest test-flags.json
# Results in empty flags object: {"flags": {}}Test error handling:
# Try to delete non-existent flag
./openfeature manifest delete non-existent --manifest test-flags.json
# Error: flag 'non-existent' not found in manifest

# Try to delete from empty manifest
echo '{"$schema": "...", "flags": {}}' > empty.json
./openfeature manifest delete any-flag --manifest empty.json
# Error: manifest contains no flags

# Try to delete from non-existent file
./openfeature manifest delete any-flag --manifest missing.json
# Error: manifest file does not existTest with debug logging:
./openfeature manifest delete feature-a --manifest test-flags.json --debug
# Should show: DEBUG Deleted flag: name=feature-a, manifestPath=test-flags.jsonView help:
./openfeature manifest delete --help
# Shows usage, examples, and flag options**3. Test Coverage Verification**

The test suite covers:
- Delete existing flag from manifest with multiple flags
- Delete the only flag in manifest (results in empty flags object)
- Error on attempting to delete non-existent flag
- Error on attempting to delete from empty manifest
- Error on missing manifest file
- Success message validation
- Manifest unchanged on error conditions
- Custom manifest path support
- Argument validation (too many/too few arguments)
- Delete first, middle, and last flags in a list

### CODE REVIEW

File: internal/cmd/manifest_delete.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+					break
+				}
+			}
+
+			if flagIndex == -1 {
+				return fmt.Errorf("flag '%s' not found in manifest", flagName)
+			}
+
+			// Remove the flag by creating a new slice without it
+			fs.Flags = append(fs.Flags[:flagIndex], fs.Flags[flagIndex+1:]...)
```
Comment (kriscoleman) [inline internal/cmd/manifest_delete.go]: I dig this one-liner, elegant!

Comment (sahidvelji) [inline internal/cmd/manifest_delete.go]:
```suggestion
			// Remove the flag
			originalLen := len(fs.Flags)
			fs.Flags = slices.DeleteFunc(fs.Flags, func(flag flagset.Flag) bool {
				return flag.Key == flagName
			})

			// Check if flag was found (length unchanged means nothing was deleted)
			if len(fs.Flags) == originalLen {
				return fmt.Errorf("flag '%s' not found in manifest", flagName)
			}
```

The zero length check can be skipped as that case will be covered by the "flag not found in manifest" check.

Comment (sahidvelji) [reply internal/cmd/manifest_delete.go]: I added a suggestion using `slices.DeleteFunc`. It allows us to get rid of the `for` loop entirely.

Comment (kriscoleman) [reply internal/cmd/manifest_delete.go]: oic, thanks!

**Comment by kriscoleman** [comment]
> Do we need to handle the case where the flag being deleted exists more than once in the manifest? Or is that impossible?

It is handled in our add command: 

https://github.com/open-feature/cli/blob/b2552286a1d2bdb22640c1e26d7fc7dc2a5640d7/internal/cmd/manifest_add.go#L167

However, it might be worth adding it to our `Validate` func. That's used by `manifest.LoadFlagSet`, which is called by all of our commands to load the flag set. 

https://github.com/open-feature/cli/blob/b2552286a1d2bdb22640c1e26d7fc7dc2a5640d7/internal/manifest/validate.go#L18
https://github.com/search?q=repo%3Aopen-feature%2Fcli%20manifest.LoadFlagSet&type=code

This way, if someone adds the flag manually by editing the json for instance, the next usage of the cli against the manifest would throw invalidation errors immediately. 

I would support that, but suggest in another PR for hygiene.

**Comment by kriscoleman** [comment]
> I would support that, but suggest in another PR for hygiene.

I added #193
